### DISCUSSION TITLE
Can I use `$this->getRelated` on already cached data?

### DISCUSSION THREAD

**Discussion by destyk** [phalcon/cphalcon#16089] [Q&A]
Good day!

Can I use `$this->getRelated` on already cached data?

For example:
1. There is a table `addresses`. All its values are cached.
2. There is a `peoples` table. Each row has an `address_id` parameter, which is tied to a specific `id` from the `addresses` table.

I am using `memcached`.

Tried like this:
```php
//...Controller

$addresses = Addresses::find();
$this->cache->set('addressesCache', $addresses, 300);

$people = Peoples::findFirst();

echo $people->getRelated('address', [
  'cache' => [
    'key' => 'addressesCache'
  ]
])->id;
```
```php
// .. Peoples model

public function initialize()
{
  $this->setSource('peoples');
  $this->belongsTo('address_id', Addresses::class, 'id', [
    'reusable' => true,
    'alias' => 'address',
    'foreignKey' => [
       'allowNulls' => true,
       'message' => 'addressId doesn\'t exists in Addresses model'
    ]
  ]);
}
```
```php
// .. Addresses model

public function initialize()
{
  $this->setSource('addresses');
  $this->hasMany('id', Peoples::class, 'address_id', [
    'alias' => 'people',
    'reusable' => true
  ]);
}
```

As a result, the `id` is displayed, which is assigned to the very first entry in the cache. But if you disable the cache, then everything works as it should.

**Comment by niden** [comment]
I don't believe this will work. Honestly I never tried it this way so I cannot be certain 100%. 

The operation is once you call getRelated()`, the query executes and if `reusable` is set, the data is stored in an internal array. Subsequent calls in the same request will give you the cached data and will not query the database again.
